vividus-plugin-rest-api/src/main/java/org/vividus/bdd/steps/api/HttpRequestSteps.java

```diff
@@ -68,9 +70,21 @@ public HttpRequestSteps(HttpTestContext httpTestContext, HttpRequestExecutor htt
      * @param content HTTP method request body
      */
     @Given("request body: $content")
-    public void request(String content)
+    public void request(DataWrapper content)
     {
-        httpTestContext.putRequestEntity(new StringEntity(content, StandardCharsets.UTF_8));
+        if (content.isDataString())
+        {
+            httpTestContext.putRequestEntity(
+                    new StringEntity(new String(content.getBytes(), StandardCharsets.UTF_8), StandardCharsets.UTF_8));
+        }
+        else
+        {
+            String resourceDataFullName = content.getResourceDataName();
+            MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create();
+            multipartEntityBuilder.addBinaryBody(resourceDataFullName, content.getBytes(), ContentType.DEFAULT_BINARY,
+                    FilenameUtils.getName(resourceDataFullName));
+            httpTestContext.putRequestEntity(multipartEntityBuilder.build());
+        }
     }
```

Comment:
my idea in pseudo-code:
```java
public void request(DataWrapper content)
{
    Object data = content.getRawData();
    HttpEntity requestEntity = data instanceof String ? new StringEntity((String) data, StandardCharsets.UTF_8) : new ByteArrayEntity((byte[]) data);
    httpTestContext.putRequestEntity(requestEntity);
}
```

Reply:
> my idea in pseudo-code:
> 
> ```java
> public void request(DataWrapper content)
> {
>     Object data = content.getRawData();
>     HttpEntity requestEntity = data instanceof String ? new StringEntity((String) data, StandardCharsets.UTF_8) : new ByteArrayEntity((byte[]) data);
>     httpTestContext.putRequestEntity(requestEntity);
> }
> ```

And we don't need any name/filename? Just bytes?

Reply:
yes, only bytes

PS: name is needed for multipart requests